terraform/main.tf

```diff
@@ -0,0 +1,550 @@
+    data.aws_subnet.cohort-8-public-subnet-2.id,
+    data.aws_subnet.cohort-8-public-subnet-3.id,
+  ]
+}
+
+# RSS ECR
+
+resource "aws_ecr_repository" "media-sentiment-rss-ecr" {
+  name         = "media-sentiment-rss-ecr"
+  force_delete = true
+}
+
+# RSS Lambda
+
+resource "aws_iam_role" "media-sentiment-rss-lambda-role" {
+  name = "media-sentiment-rss-lambda-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Action = "sts:AssumeRole",
+        Effect = "Allow",
+        Principal = {
+          Service = "lambda.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_lambda_function" "media-sentiment-rss-lambda" {
+  function_name = "media-sentiment-rss-lambda"
+  role          = resource.aws_iam_role.media-sentiment-rss-lambda-role.arn
+  image_uri     = "129033205317.dkr.ecr.eu-west-2.amazonaws.com/media-sentiment-rss-ecr:latest"
+  architectures = ["x86_64"]
+  package_type  = "Image"
+  timeout = 120
+  environment {
+    variables = {
+      INITIAL_DATABASE  = var.initial_database
+      DATABASE_NAME     = var.database_name
+      DATABASE_PASSWORD = var.database_password
+      DATABASE_PORT     = var.database_port
+      DATABASE_USERNAME = var.database_username
+      DATABASE_IP       = aws_db_instance.media-sentiment-rds.address
+      ACCESS_KEY_ID     = var.access_key
+      SECRET_ACCESS_KEY = var.secret_key
+    }
+  }
+}
+
+# RSS Scheduler
+
+resource "aws_iam_role" "media-sentiment-rss-scheduler-role" {
+  name = "media-sentiment-rss-scheduler-role"
+  assume_role_policy = jsonencode({
+  Version = "2012-10-17",
+  Statement = [
+    {
+      Effect = "Allow",
+      Principal = {
+        Service = "scheduler.amazonaws.com"
+      },
+      Action = "sts:AssumeRole"
+    }
+  ]
+})
+  inline_policy {
+    name = "media-sentiment-rss-inline-policy"
+
+    policy = jsonencode({
+	Version= "2012-10-17",
+	Statement = [
+		{
+			Effect= "Allow",
+			Action= "lambda:InvokeFunction"
+      Resource= [
+                "arn:aws:lambda:eu-west-2:129033205317:function:media-sentiment-rss-lambda:*",
+                "arn:aws:lambda:eu-west-2:129033205317:function:media-sentiment-rss-lambda"
+            ]
+		},
+    ]
+})
+}
+}
+
+resource "aws_scheduler_schedule" "media-sentiment-rss-pipeline-scheduler" {
+  name                         = "media-sentiment-rss-pipeline-scheduler"
+  schedule_expression_timezone = "Europe/London"
+  description                  = "Schedule to run ETL pipeline every hour"
+  state                        = "ENABLED"
+  
+  flexible_time_window {
+    mode = "OFF"
+  }
+
+  schedule_expression = "cron(0 * * * ? *)"
+
+  target {
+    arn      = "arn:aws:lambda:eu-west-2:129033205317:function:media-sentiment-rss-lambda"
+    role_arn = "arn:aws:iam::129033205317:role/media-sentiment-rss-scheduler-role"
+  }
+}
+
+# Media Sentiment RDS
+
+resource "aws_security_group" "media-sentiment-rds-sg" {
+  vpc_id = "vpc-0e0f897ec7ddc230d"
```

Comment:
Is it possible to specify this by referring to the resource rather than a hard-coded string?